apps/src/music/views/Controls.tsx

```diff
@@ -117,6 +119,20 @@ const Controls: React.FunctionComponent<ControlsProps> = ({
     state => state.music.soundLoadingProgress < 1
   );
 
+  // Sends a Statsig event when the Run button is pressed when a user is signed out
+  const sendAnalyticsEvent = (): void => {
```

Comment:
I'd recommend firing this event in `MusicView.jsx`, here: 
https://github.com/code-dot-org/code-dot-org/blob/84c4ba1aed261d0371ffff1a4931c728ca4da91b/apps/src/music/views/MusicView.jsx#L500-L509
Since we already fire another analytics event there (to the Music Lab amplitude project), and we should have access to the signed in state via `this.props.signInState`

Reply:
This was super helpful, thank you!

I'm going to move this change and the other Run button change into a separate PR since these are more complicated than the others. Is it ok if I tag you for review in the new PR when it's ready?